lib/openstudio-standards/standards/Standards.PlantLoop.rb

```diff
@@ -1495,4 +1495,43 @@ def plant_loop_capacity_w_by_maxflow_and_delta_t_forwater(plant_loop)
     plantloop_capacity = plantloop_dt * plantloop_maxflowrate * 1000.0 * 4180.0
     return plantloop_capacity
   end
+
+  # This methods replaces all indoor or outdoor pipes which model the heat transfer between the pipe and the
+  # environement by adiabatic pipes.
+  #
+  # @param plant_loop [OpenStudio::Model::PlantLoop] plant loop
+  # @return [Bool] returns true if successful
+  def plant_loop_adiabatic_pipes_only(plant_loop)
+    supply_side_components = plant_loop.supplyComponents
+    demand_side_components = plant_loop.demandComponents
+    plant_loop_components = supply_side_components += demand_side_components
+    plant_loop_components.each do |component|
+      # Get the object type
+      obj_type = component.iddObjectType.valueName.to_s
+      next if !['OS_Pipe_Indoor', 'OS_Pipe_Outdoor'].include?(obj_type)
+
+      # Get pipe object
+      pipe = nil
+      case obj_type
+      when 'OS_Pipe_Indoor'
+        pipe = component.to_PipeIndoor.get
+      when 'OS_Pipe_Outdoor'
+        pipe = component.to_PipeOutdoor.get
+      end
+
+      # Get pipe node
+      node = prm_get_optional_handler(pipe, @sizing_run_dir, 'to_StraightComponent', 'outletModelObject', 'to_Node')
```

Comment:
why not use the outletModelObject field from the StraightComponent parent class instead?

Reply:
That lines is equivalent to `pipe.to_StraightComponent.outletModelObject.to_Node`, so we do use the `outletModelObject` from the `StraightComponent` parent class. Could you please clarify what you mean?

Reply:
just replace it with node = pipe.to_StraightComponent.outletModelObject.to_Node

also not sure node_name is used in the following lines?

Reply:
Doesn't `to_Node` return an optional?